### ISSUE TITLE
[Bug]: plot directive not handling indentation of long option strings

### ISSUE THREAD

**Issue by story645** [matplotlib/matplotlib#29649]
### Bug summary

In  #29644, I couldn't wrap alt text because the alt field wasn't getting properly parsed by the plot directive. Instead it yeilded something like the following for every variant 

### Code for reproduction

```rst
.. plot:: _embedded_plots/hatch_classes.py
    :alt: 8 squares, each showing the pattern corresponding to the hatch symbol: symbol
        '/' makes right leaning diagonals, '\\' makes left leaning diagonals, '|' makes
        vertical lines, '-' makes horizontal lines, '+' makes a grid, 'X' makes a grid
        rotated 90 degrees, 'o' makes small unfilled circles, 'O' makes large unfilled
        circles, '.' makes small filled circles,
        and '*' makes a start with 5 points
    :include-source: false
```

### Actual outcome

![Image](https://github.com/user-attachments/assets/913e7be5-da52-4c28-869a-4a83c4da58ab)

### Expected outcome

The alt text properly parsed so just the image shows and the alt txt is put into the html

### Matplotlib Version

dev

**Comment by anntzer** [comment]
This is a rst field list, I suspect (but haven't checked) that the correct way to handle multilines is to also indent the first line, see https://docutils.sourceforge.io/docs/user/rst/quickref.html#field-lists (see `:Authors:` in the example and the remark "Field lists are used as part of an extension syntax, such as options for directives etc.")

**Comment by tacaswell** [comment]
I only took a very quick look, but I don't think we are doing any of the parsing ourselves:

https://github.com/matplotlib/matplotlib/blob/177f28cd566d77dfac83f859e0ce9743cbe995cc/lib/matplotlib/sphinxext/plot_directive.py#L254-L274

we are leaving that to sphinx/docutils.  Maybe adding a line continuation (`\`) will work?  If not I think this needs to go to sphinx and/or docutils.

**Comment by timhoffm** [comment]
Try an `.. image::` directive with `:alt:`. That's a very basic docutils directive using the same definition as we do
https://github.com/docutils/docutils/blob/f704fb58904d62bf5c6d7db82a9628f29e4ed246/docutils/docutils/parsers/rst/directives/images.py#L50

If that does not work either, it's a docutils issue - either a bug or an incorrect documentation.

**Comment by QuLogic** [comment]
It is strange, because [`Image` defines `alt` as `unchanged`](https://sourceforge.net/p/docutils/code/HEAD/tree/trunk/docutils/docutils/parsers/rst/directives/images.py#l50), but so do we:
https://github.com/matplotlib/matplotlib/blob/0b7a88a967ce8d02284dd4c7e47889ccc0bbcbf0/lib/matplotlib/sphinxext/plot_directive.py#L262

So there should be no difference in parsing.

**Comment by QuLogic** [comment]
It must be a bug with `has_content = True` for the directive (since `image` doesn't have content, but `plot` does.)

**Comment by QuLogic** [comment]
For reasons I don't quite understand, the plot directive renders the plots, and then _templates it out into new reST_, which is inserted into the parser after it. This path through text again doesn't handle multi-line input and is what causes the breakage.
